### PR TITLE
pci: allow bus Read() to take a ...Filter

### PR DESCRIPTION
**Pull request by rminnich**
Sometimes we wish to block, match, reject devices based on some
criteria.

If this meets approval I'll add a few examples as tests. We could for example filter
on vendor id as string, name as string, type, etc.

### CODE REVIEW

File: pkg/pci/types.go [hunk 1]
```diff
@@ -4,8 +4,11 @@
 
 package pci
 
+// Filter can be used to filter a device
+type Filter func(p *PCI) bool
+
 type busReader interface {
```
Comment (rjoleary) [inline pkg/pci/types.go:10]: Is there a reason `busReader` is not exported? It feels a bit weird that `NewBusReader()` is returning something not exported. It's also referred to as `BusReader` in the documentation.

Comment (rminnich) [reply pkg/pci/types.go:10]: that's in a later PR but yeah that needs fixing. There's a bunch of little bits to fix in this thing.

**Comment by GanShun** [comment]
Is this meant to help https://github.com/u-root/u-root/pull/1768 ?
If this is a general filter then yes, but it doesn't help what I'm trying to do in 1768

**Comment by rminnich** [comment]
you're right Gan but I figure this is still worth doing.

**Pull request merged**